components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java

```diff
@@ -174,7 +176,18 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) {
                 String serviceProviderName = null;
                 String serviceProviderUUID = null;
                 try {
-                    serviceProvider = OAuth2Util.getServiceProvider(oAuth2IntrospectionResponseDTO.getClientId());
+                    /*
+                     Tokens which are issued for the applications which are registered in sub organization,
+                     contains the tenant domain for the authorized user as the sub organization. Based on that
+                     we can get the application details by using both the client id and the tenant domain.
+                    */
+                    if (OrganizationManagementUtil.isOrganization(oAuth2IntrospectionResponseDTO.getAuthorizedUser().
```

Comment:
oAuth2IntrospectionResponseDTO.getAuthorizedUser() has been assigned to authorizedUser variable above.
we can use authorizedUser.getTenantDomain() to a variable and use it in following places. There are multiple usages of oAuth2IntrospectionResponseDTO.getAuthorizedUser().getTenantDomain()

Reply:
Done